option.go

```diff
@@ -194,6 +194,22 @@ func WithSchema(schema any) Option {
 	return &schemaOption{Schema: schema}
 }
 
+// InitializerFunc is a function that initializes a message. It may be used to
+// dynamically construct messages. It is called on client and handler Receive to
+// construct the message to be unmarshaled into.
+type InitializerFunc func(spec Spec, msg any) error
+
+// WithInitializer provides a function that initializes a message. It may be
+// used to dynamically construct messages.
+//
+// By default, an initializer is provided to support [dynamicpb.Message]
+// messages. This initializer sets the descriptor from the Schema field of the
+// [Spec]. The Schema must be of type [protoreflect.MethodDescriptor] to use
+// dynamicpb.Message.
+func WithInitializer(initializer InitializerFunc) Option {
```

Comment:
Splitting into two might make it clearer, wanted to try the single version but will switch to:
```go
func WithResponseInitializer(initializer InitializerFunc) ClientOption
func WithRequestInitializer(initializer InitializerFunc) HandlerOption
```

Reply:
Split into two.